automation/ansible/roles/setup_jmeter_client/tasks/main.yml

```diff
@@ -0,0 +1,59 @@
+---
+- name: Install jmeter client packages.
+  become: yes
+  become_user: root
+  ansible.builtin.package:
+    name: "{{ jmeter_packages }}"
+    state: present
+
+- name: Prepare JMeter client directory
+  shell: |
+    if [ ! -d "{{ jmeter_remote_bin_dir_full_path }}" ]; then
+      mkdir -p "{{ jmeter_remote_bin_dir_full_path }}"
+    fi
+  register: prepare_jmeter_dir_output
+
+- name: "Download jmeter binary {{ jmeter_binary_archive }}"
+  get_url:
+    url: "{{ jmeter_binary_download_uri }}/{{ jmeter_binary_archive }}"
+    dest: "{{ jmeter_remote_bin_dir_full_path }}/{{ jmeter_binary_archive }}"
+    mode: '0744'
```

Comment:
Mode should be 644 as this is a zip file

Reply:
Made the changes and I will be checking in soon.